### PR TITLE
chore: move to clang-format 1.0.17.

### PR DESCRIPTION
**Pull request by mprobst**
clang-format 1.0.17 substantially improves formatting for fat arrow functions
and array literal detection. It also fixes a number of minor formatting issues.

### CODE REVIEW

File: modules/angular2/test/core/compiler/dynamic_component_loader_spec.ts [hunk 1]
```diff
@@ -76,26 +76,24 @@ export function main() {
...
-                                          view.context.ctxBoolProp = true;
-                                          view.detectChanges();
-
-                                          var dynamicComponent =
-                                              view.rawView.viewContainers[0].views[0].locals.get(
-                                                  "dynamic");
-                                          return dynamicComponent.done;
-                                        })
+             dynamicComponent.done.then((_) => {
+                                    view.detectChanges();
```
Comment (rkirov) [inline modules/angular2/test/core/compiler/dynamic_component_loader_spec.ts]: interesting choice of indentation here.

File: modules/benchpress/src/sample_description.ts [hunk 2]
```diff
@@ -23,22 +23,23 @@ export class SampleDescription {
...
+var _BINDINGS = [
+  bind(SampleDescription)
+      .toFactory((metric, id, forceGc, userAgent, validator, defaultDesc, userDesc) =>
+                         new SampleDescription(id, [
+                           {'forceGc': forceGc, 'userAgent': userAgent},
+                           validator.describe(),
+                           defaultDesc,
+                           userDesc
+                         ],
+                                               metric.describe()),
```
Comment (rkirov) [inline modules/benchpress/src/sample_description.ts]: lol

File: npm-shrinkwrap.clean.json [hunk 3]
```diff
@@ -4270,7 +4270,7 @@
           }
         },
         "clang-format": {
-          "version": "1.0.15"
+          "version": "1.0.17"
```
Comment (rkirov) [inline npm-shrinkwrap.clean.json]: I assume you did this by hand, ideally we should bump the gulp-clang-format and receive the new version here as a byproduct. But I don't blame you give the 'fun' I am having with shrinkwrap.

File: modules/benchpress/src/sample_description.ts (hunk 2 above)
Comment (mprobst) [inline modules/benchpress/src/sample_description.ts]: Yeah, known issue. It was like that before, too, so it's at least not a regression.

File: npm-shrinkwrap.clean.json (hunk 3 above)
Comment (mprobst) [inline npm-shrinkwrap.clean.json]: Ack.

File: modules/angular2/test/core/compiler/dynamic_component_loader_spec.ts (hunk 1 above)
Comment (mprobst) [inline modules/angular2/test/core/compiler/dynamic_component_loader_spec.ts]: Yeah, it's caused by the subsequent .then(). I extracted a var, looks better now.

**Comment by mprobst** [comment]
This is fixed at HEAD.

**Pull request closed without merging**